app/controllers/RestApi.scala

```diff
@@ -87,6 +101,118 @@ class RestApi @Inject() (
 
   }}
 
+  case class CreateUserBody(
+     email: String,
+     firstName : String,
+     lastName: String,
+     mobilePhone: String,
+     placeOfResidence: String,
+     birthday: Long,
+     sex: String,
+     password: String,
+     profileImageUrl: Option[String]
+  )
+  object CreateUserBody{
+    implicit val createUserBodyJsonFormat = Json.format[CreateUserBody]
+  }
+
+  def createUser() = ApiAction.async(validateJson[CreateUserBody]) { implicit request => {
+    val signUpData = request.request.body
+
+    val loginInfo = LoginInfo(CredentialsProvider.ID, signUpData.email)
+    userService.retrieve(loginInfo).flatMap{
+      case Some(_) =>
+        Future(BadRequest(Json.obj("error" -> Messages("error.userExists", signUpData.email))))
+      case None =>{
+        val profile = Profile(loginInfo, true, signUpData.email, signUpData.firstName, signUpData.lastName, signUpData.mobilePhone, signUpData.placeOfResidence, signUpData.birthday, signUpData.sex, List(new DefaultProfileImage))
+        avatarService.retrieveURL(signUpData.email).flatMap(avatarUrl  => {
+          userService.save(User(id = UUID.randomUUID(), profiles =
+            (signUpData.profileImageUrl, avatarUrl) match {
+              case (Some(url), Some(gravatarUrl))  => List(profile.copy(avatar = List(UrlProfileImage(url),GravatarProfileImage(gravatarUrl),new DefaultProfileImage)))
+              case (None, Some(gravatarUrl))=> List(profile.copy(avatar = List(GravatarProfileImage(gravatarUrl),new DefaultProfileImage)))
+              case (Some(url), None) => List(profile.copy(avatar = List(UrlProfileImage(url), new DefaultProfileImage)))
+              case _ => List(profile.copy(avatar = List(new DefaultProfileImage)))
+            })).flatMap((user) => {
+              authInfoRepository.add(loginInfo, passwordHasher.hash(signUpData.password)).map(_ => Ok(Json.toJson(user)))
+          })
+        })
+      }
+    }
+  }}
+
+  case class UpdateUserBody(
+                             email: String,
+                             firstName : Option[String],
+                             lastName: Option[String],
+                             mobilePhone: Option[String],
+                             placeOfResidence: Option[String],
+                             birthday: Option[Long],
+                             sex: Option[String],
+                             password: Option[String]
```

Comment:
I'm not sure, but it seems to me that notings happens with the password attribute of UpdateUserBody. In my opinion, we should not allow to change the password of a users account by external tools. Therefore it has to be removed.